src/Faker/ValidGenerator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Faker;
+
+/**
+ * Proxy for other generators, to return only valid values. Works with
+ * Faker\Generator\Base->valid()
+ */
+class ValidGenerator
+{
+    protected $generator;
+    protected $validator;
+    protected $maxRetries;
+
+    /**
+     * @param Generator $generator
+     */
+    public function __construct(Generator $generator, $validator = null, $maxRetries = 10000)
```

Comment:
I want `$validator` to have a default value of accepting all values (the true function), and PHP only accepts scalar values as default arguments.

I added a check later to make sure the parameter is a callable.